DOCUMENTATION.md

```diff
@@ -43,6 +45,190 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
 
 ![Propulsion c4model.com Container Diagram: Reactors](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.github.com/jet/propulsion/master/diagrams/ReactorsContainer.puml&fmt=svg)
 
+# The `Propulsion.Streams` Programming Model for Projections, Reactions and Workflows
+
+`Propulsion.Streams` provides a programming model to manage running of _Handlers_ in a manner that optimises for the following:
```

Comment:
**SUGGESTION:** I think that it'd be great to add, at least, the general mission statement on top of the documentation or here. So what issue is Propulsion trying to fix, and what're streams and handlers in this context. I know that there's a glossary in the main doc, but even to support the learning cycle, repetition is good. Quite often people are skimming first searching for the term they're interested, then they're trying to look broader.

Reply:
Good call...
- [Partially addressed by this rewording](https://github.com/jet/propulsion/pull/200/commits/3cf75c14d1411bce030d39192b5d7995b9350a9b).
- Also, there's now a glossary.

I don't feel the full spirit of your suggestion has been addressed as such, but I'm going to leave it at this for now (the README should ultimately be providing some of this intro stuff; this programming guide should probably put a slightly different emphasis on the same general theme and work together)